backend/src/main/java/com/redhat/cloud/notifications/events/FromCamelHistoryFiller.java

```diff
@@ -70,6 +96,65 @@ void init() {
                 .eventually(() -> messagesProcessedCounter.increment());
     }
 
+    private Uni<Void> reinjectIfNeeded(Map<String, Object> payload) {
+        if (!enableReInject || (payload.containsKey("successful") && ((Boolean) payload.get("successful")))) {
+            return Uni.createFrom().voidItem();
+        }
+
+        String historyId = (String) payload.get("historyId");
+        log.infof("Notification with id %s was not successful, resubmitting for further processing", historyId);
+
+        Uni<Endpoint> endpointUni = notificationResources.getEndpointForHistoryId(historyId);
+
+        return endpointUni.invoke(ep -> {
+
+            Event event = new Event();
+
+            // TODO augment with details from Endpoint and original event
+            event.setPayload(payload);
+
+            // Save the original id, as we may need it in the future.
+            Map<String, String> context = new HashMap<>();
+            context.put("original-id", historyId);
+            if (ep != null) { // TODO For the current tests. EP should not be null in real life
+                context.put("failed-integration", ep.getName());
+            }
+
+            Action action = Action.newBuilder()
+                    .setBundle("platform")
+                    .setApplication("notifications")
+                    .setEventType("integration_failed")
```

Comment:
> Wrt the EventTypeName , I have no opinion @josejulio should probably decide.

+1 on using `integration-failed`

```suggestion
                    .setEventType("integration-failed")
```